test/metabase/query_processor_test/cast_test.clj

```diff
@@ -496,25 +499,22 @@
 
 ;; date()
 
-(deftest ^:parallel date-parse-table-fields
-  (mt/test-drivers (mt/normal-drivers-with-feature :expressions/date)
-    (let [mp (mt/metadata-provider)]
-      (doseq [[table fields] [[:people [{:field :birth_date}]]]
-              {:keys [field]} fields]
-        (testing (str "casting " table "." field " to date")
-          (let [field-md (lib.metadata/field mp (mt/id table field))
-                query (-> (lib/query mp (lib.metadata/table mp (mt/id table)))
-                          (lib/with-fields [field-md])
-                          (lib/expression "DATECAST" (lib/date (lib/text field-md)))
-                          (lib/limit 100))
-                result (-> query qp/process-query)
-                cols (mt/cols result)
-                rows (mt/rows result)]
-            (is (types/field-is-type? :type/Date (last cols)))
-            (doseq [[uncasted-value casted-value] rows]
-              (let [cd (-> casted-value java.time.Instant/parse)
-                    ud (-> uncasted-value java.time.Instant/parse)]
-                (is (= ud cd))))))))))
+(defn- date-type? [col]
+  (some #(types/field-is-type? % col) [:type/DateTime ;; some databases return datetimes for date (e.g., Oracle)
+                                       :type/Text ;; sqlite uses text :(
+                                       :type/Date
+                                       :type/* ;; Mongo
```

Comment:
Maybe it makes sense to just remove the assertion at this point?

Reply:
Yeah, that might be a good move. I'd rather leave it if only as a comment listing the expectations. Someone might figure out why Mongo returns * for this but not for the coercion.